src/swh_api_functions.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import requests
+
+class CustomApiResponse:
+    """
+    this class represents response data structure for returning response
+    from functions.
+    """
+
+    def __init__(self, data):
+        self.checksums = None
+        self.data_url = None
+        self.filetype_url = None
+        self.language_url = None
+        self.length = None
+        self.license_url = None
+        self.status = None
+
+        for key in vars(self):
+            if key in data:
+                setattr(self, key, data[key])
+
+    def __eq__(self, second_class_obj):
```

Comment:
@pombredanne  If you are asking about __eq__, its required when i compare two objects while testing so that it compares each and every attribute of two objects.
And if you are asking about `for key in vars(self)` this loop, its required to set attributes as of now becuase there are lot of attributes so its difficult to pass each of'em in contructor and set them bcoz of sequence and all.
Thanks.